File: my-submissions/2428-equal-row-and-column-pairs/solution.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var equalPairs = function (grid) {
    const n = grid.length;
    const rowMap = new Map();

    for (const row of grid) {
        const key = row.join(',')
        rowMap.set(key, (rowMap.get(key) || 0) + 1);

    }

    // console.log(rowMap);
    let count = 0;

    for (let col = 0; col < n; col++) {
        const colArr = [];

        for (let row = 0; row < n; row++) {
            colArr.push(grid[row][col])
        }

        // console.log(colArr);
        const key = colArr.join(',');
        if (rowMap.has(key)) {
            count += rowMap.get(key);
        }
    }
    return count;
};

File: my-submissions/2428-equal-row-and-column-pairs/solution.java
class Solution {
    public int equalPairs(int[][] grid) {

        int n = grid.length;
        Map<String, Integer> rowMap = new HashMap<>();

        for(int[] row: grid){
           String key = Arrays.toString(row);
           rowMap.put(key, rowMap.getOrDefault(key, 0) + 1);
        }

        int count = 0;

        // System.out.println(rowMap);

        // compare col with row
        for(int j = 0; j < n; j++){
            int[] col = new int[n];
            for(int i = 0; i < n; i++){
                col[i] = grid[i][j];
            }
            String key = Arrays.toString(col);
            count += rowMap.getOrDefault(key, 0);
        }
        
        return count;
    }
}

